internal/curator/durable/state/fb/state.fbs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2017 Western Digital Corporation or its affiliates. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+namespace internal.curator.durable.state.fb;
+
+// TractIDF is a 10-byte value that can refer to a tract or an RS chunk, just like common.TractID.
+// Declare it with shorts so flatbuffers doesn't pad for excessive alignment.
+struct TractIDF {
+  b1: ushort;
+  b2: ushort;
+  b3: ushort;
+  b4: ushort;
+  b5: ushort;
+}
+
+// PartitionF holds data about one partition. ID is not strictly necessary since it should
+// always be determined by context, but it makes some code simpler, and there will be very
+// few of these.
+table PartitionF {
+  id: uint;
+  next_blob_key: uint;
+  next_rs_chunk_key: ulong; // 48 bits of "blob key" and "tract index" in one value
+}
+
+// TractF holds data about one tract of one blob.
+table TractF {
+  // Used for REPLICATED class.
+  hosts012: ulong;  // first three hosts packed into a ulong, 20 bits each
+  hosts3p: [uint];  // hosts beyond the first three
+
+  // Used for all classes.
+  version: uint;
+
+  // Used for RS_X classes.
+  // There must be exactly one of these for each RS_n_m enum value in
+  // common.StorageClass.
+  rs63_chunk:  TractIDF;
+  rs83_chunk:  TractIDF;
+  rs103_chunk: TractIDF;
+  rs125_chunk: TractIDF;
+}
+
+// BlobF holds data about one blob.
+table BlobF {
+  // To save space in the flatbuffers encoding, we pack several small fields
```

Comment:
I think the vtable overhead is 2 bytes per field, so 1 uint uses 6 bytes, and 3 ubytes use 9. But there's alignment also, so I think actually it might be more like 12.

I did a bunch of experiments a while ago taking data from an old curator database and seeing which representations came out the smallest. Unfortunately I lost that code (and data), but it shouldn't be that hard to write a few quick experiments.

I was trying to find a balance between excessive complexity and saving space... there are some more tricks that could be applied to save a few more bytes (in RSChunk in particular), but after the major wins (which made it into this diff), it felt like it was hitting diminishing returns.